lib/asciidoctor/opal_ext/file.rb

```diff
@@ -19,12 +19,19 @@ class File
   attr_reader :lineno
   attr_reader :path
 
-  def initialize(path, mode = 'r')
+  def initialize(path, flags = 'r')
     @path = path
     @contents = nil
     @eof = false
     @lineno = 0
-    @fd = `require('fs').openSync(path, mode)` if JAVASCRIPT_PLATFORM == 'node'
+    # binary flag is unsupported
+    binary_flag_regexp = /b/
```

Comment:
I think a better approach here is to use `delete`. No need to be diving into regex.

```ruby
flags = (flags.delete 'b')
```

If you want to check if it was present first, add an `include?`

```ruby
binary = false
if flags.include? 'b'
  binary = true
  flags = flags.delete 'b'
end
```

Always try to find a way to do it without regex first.